tests/Services/Search/ParametersHandlerTest.php

```diff
@@ -0,0 +1,318 @@
+                ],
+                'texts.name1_asc'
+            ],
+            '$isSearch is false and defaultSorting is item.random' => [
+                false,
+                'item.random',
+                [
+                    'default.recommended_sorting',
+                    'texts.name1_asc',
+                    'texts.name1_desc',
+                    'sorting.price.avg_asc',
+                    'sorting.price.avg_desc',
+                    'variation.createdAt_desc',
+                    'variation.createdAt_asc',
+                    'variation.availability.averageDays_asc',
+                    'variation.availability.averageDays_desc',
+                    'variation.number_asc',
+                    'variation.number_desc',
+                    'variation.updatedAt_asc',
+                    'variation.updatedAt_desc',
+                    'item.manufacturer.externalName_asc',
+                    'item.manufacturer.externalName_desc',
+                    'variation.position_asc',
+                    'variation.position_desc',
+                    'item.random'
+                ],
+                [
+                    'texts.name1_asc' => 'Ceres::Template.itemName_asc',
+                    'texts.name1_desc' => 'Ceres::Template.itemName_desc',
+                    'sorting.price.avg_asc' => 'Ceres::Template.itemPrice_asc',
+                    'sorting.price.avg_desc' => 'Ceres::Template.itemPrice_desc',
+                    'variation.createdAt_desc' => 'Ceres::Template.itemVariationCreateTimestamp_desc',
+                    'variation.createdAt_asc' => 'Ceres::Template.itemVariationCreateTimestamp_asc',
+                    'variation.position_asc' => 'Ceres::Template.itemVariationTopseller_asc',
+                    'variation.position_desc' => 'Ceres::Template.itemVariationTopseller_desc'
+                ],
+                'item.score'
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider providerHandlePaginationAndSorting
+     *
+     * @param $isSearch
+     * @param $defaultSortingSearch
+     * @param $configData
+     * @param $sortingOptions
+     * @param $defaultOption
+     */
+    public function testHandlePaginationAndSorting($isSearch, $defaultSortingSearch, $configData, $sortingOptions, $defaultOption)
+    {
+        /** @var ExternalSearchOptions|MockObject $externalSearchOptionsMock */
+        $externalSearchOptionsMock = $this->getMockBuilder(ExternalSearchOptions::class)
+            ->setMethods(null)
+            ->getMock();
+
+        /** @var Request|MockObject $requestMock */
+        $requestMock = $this->getMockBuilder(Request::class)
+            ->setMethods([])
+            ->getMock();
+
+        if ($isSearch) {
+            $requestMock->expects($this->once())->method('getUri')->willReturn('test.com/search');
+        }
+
+        $config = $this->createCeresConfigObject(
+            $configData,
+            $defaultSortingSearch,
```

Comment:
Rename this to `$defaultSorting` since it could also contain the sorting for category pages.